pycbc/detector.py

```diff
@@ -106,27 +111,23 @@ def add_detector_on_earth(name, longitude, latitude,
         # assume right angle detector if no separate xarm direction given
         xangle = yangle + np.pi / 2.0
 
-    # Rotation matrix to move detector to correct orientation
-    rm1 = rotation_matrix(longitude * units.rad, 'z')
-    rm2 = rotation_matrix((np.pi / 2.0 - latitude) * units.rad, 'y')
-    rm = np.matmul(rm2, rm1)
-
+    # baseline response of a single arm pointed in the -X direction
+    resp = np.array([[-1, 0, 0], [0, 0, 0], [0, 0, 0]])
+    rm2 = rotation_matrix(-longitude * units.rad, 'z')
+    rm1 = rotation_matrix(-1.0 * (np.pi / 2.0 - latitude) * units.rad, 'y')
+    
     # Calculate response in earth centered coordinates
     # by rotation of response in coordinates aligned
     # with the detector arms
     resps = []
     vecs = []
-    for angle in [yangle, xangle]:
-        a, b = cos(2 * angle), sin(2 * angle)
-        resp = np.array([[-a, b, 0], [b, a, 0], [0, 0, 0]])
-
+    for angle, azi in [(yangle, yaltitude), (xangle, xaltitude)]:
+        rm0 = rotation_matrix(angle * units.rad, 'z')
+        rmN = rotation_matrix(-azi *  units.rad, 'y')
+        rm = rm2 @ rm1 @ rm0 @ rmN
         # apply rotation
-        resp = np.matmul(resp, rm)
-        resp = np.matmul(rm.T, resp) / 4.0
-        resps.append(resp)
-
-        vec = np.matmul(rm.T, np.array([-np.cos(angle), np.sin(angle), 0]))
-        vecs.append(vec)
+        resps.append(rm @ resp @ rm.T / 2.0)
+        vecs.append(rm @ np.array([-1, 0, 0]))
```

Comment:
So there's changes in here. I think to include the "altitude angles" of the two arms. Have you checked/can you show that this is unchanged in the limit that the altitude angles are both 0.

For current instruments (H1,L1,V1,K1), will this make any actual change?

Reply:
@spxiwh That's actually what I added as a unit test. The key thing is that this sets up the response matrix. I numerically compare the response matrix here against the cached ones in lal. There is no change to the response there (except the E* case which has a 10^-4 level change due to the issue I raised in slack). 

The altitudes are near zero for the H1/ L1 / V1 comparison, but not exactly so (nor should they be). In fact if you don't account for the altitude the response matrix would also be different at the O(10^-4) level like E* is.